[#41849] adapt the style of the attachments section to the new designs

Kharonus opened the pull request:
- https://community.openproject.org/work_packages/41849

Kharonus committed:
[#41849] Adapt the style of the attachments section to the new designs

- https://community.openproject.org/work_packages/41849

Kharonus committed:
[#41849] add data-qa-selector to new attachment html

Kharonus committed:
[#41849] added some spacings after QA of UIX

HDinger reviewed (changes requested):
There are some discrepancies between the things defined in the linked ticket and the implementation. The biggest one is probably the file information section, which is probably out of scope(?, see below). If yes, please consider updating the ticket.

- [x] The file icon still shows the paper clip and not the type-dependent icon
- [x] As a user with the role "Reader", I see the delete icon on hover. When I click it, nothing happens, but it should not be visible at all.
- [x] When hovering, I see a user icon which leads me to the creator of the attachments. That felt very weird, and I did not know what to expect from this icon. In the specification, I could not find any information about this. Do we really need it?
- [x] According to the specification, the first element in the list, should be the newest one. Currently it seems to be alphabetical, but only after a reload. When I upload an attachment, it is glued to the bottom.
<img width="300" alt="Bildschirmfoto 2022-06-20 um 11 36 18" src="https://user-images.githubusercontent.com/7457313/174573091-6296b54d-5e7a-4398-aff5-5a70ea0beccd.png">

 
- [x] The ticket specifies a file information section, but I assume that is out of scope for this ticket?

Kharonus commented:
Many thanks for your review.

-> clip icon is wrongly used for the attachments itself. yes, this is apparently due to a complete confusion on my part. rechecked specification, and will change those icons

-> delete icon must be hidden, will do so

-> author action: this was introduced by me as a proactive decision. It is based on the fact, that we hide author name and link behind a not clickable (and not hoverable) avatar now. So, if you are still interested in the author, you can now hover the author icon and even use the action to go to the author's page. Talked to marc about it, and until we have a better solution (or the decision to willingly drop that information about an attachments author), we can keep it. Maybe even to get some feedback on it.

-> sorting: I will change it, was by file name before the refactoring though, I didn't change it in this PR.

-> file information section is out of scope, correct.

Kharonus committed:
[#41849] added sorting to attachments list

- replaced angular trackBy with own sorting inside observable
- added file icons to attachment list items
- fixed clip icon usage

HDinger reviewed (approved):
👍
